proto/p4/p4runtime.proto

```diff
@@ -136,6 +136,14 @@ message TableEntry {
   // this is stored, as long as it is returned with the rest of the entry in
   // a Read RPC.
   uint64 controller_metadata = 5;
+  // meter_config and counter_data are convenience fields that enable the
+  // controller to configure the direct resources associated with the table at
+  // the same time as a match-action entry is inserted or modified. When
+  // updating a table entry, leaving these fields unset means that the direct
+  // resources (if any) will not be updated. When reading a table entry, the
+  // values associated with the direct resources (if any) will be returned.
```

Comment:
I believe the intent is to return a value for the direct resource field in a table read if the direct resource is specified for the table. Added another comment for initializing the direct resource field(s), which should cover your concern.